Systick refactor

micromouseonline opened the pull request:
Systick now runs at 20kHz and samples all six sensor channels as well as the battery and function switch. 

Updates have been made to the sensor display commands as well so that all six channels are displayed. 

Also tucked in is a small change to the hex display of channel data just for a bit of clarity.

As far as I can tell, the outcomes are still the same. 

If you have not already noted this elsewhere, unconnected ADC channels will display a value dependent upon the most recently sampled channel. this could be eliminated in software by setting unused channels to outputs and writing a zero. I think.

micromouseonline committed:
Merge pull request #2 from micromouseonline/commas-as-characters

also send the changed marker as char

micromouseonline committed:
Merge branch 'master' of https://github.com/robzed/ukmarsey

micromouseonline committed:
Merge branch 'master' of https://github.com/robzed/ukmarsey

micromouseonline committed:
first draft new systick

micromouseonline committed:
Systick runs at 20kHz - sample all channels

Also update sensor commands to show all six channels

robzed commented on sensors_control.cpp:
```diff
@@ -105,73 +109,108 @@ void sensors_control_setup() {
+    Serial.print(a5_lit);
   }
   Serial.println();
+  delay(100);
```
Why is there a delay 100? Left over from testing?

micromouseonline replied:
Yes, my mistake. There is also the looped nature of the response to take out. I will do that later today.

robzed commented on systick.cpp:
```diff
@@ -110,23 +156,126 @@ unsigned long t_systick1 = 0;
+ * purpose.
+ */ 
+ISR(TIMER2_COMPA_vect) {
+  digitalWriteFast(LED_BUILTIN, 1);
```
Is this a testing line?

micromouseonline replied:
I can remove the LED flashing. It is just for timing with an oscilloscope.

robzed commented on systick.cpp:
```diff
@@ -110,23 +156,126 @@ unsigned long t_systick1 = 0;
+    default:
+      break;
+  }
+  digitalWriteFast(LED_BUILTIN, 0);
```
Testing line?

micromouseonline replied:
As for previous LED flashing - it can go.

robzed reviewed (changes requested):
Changes are excellent. I want to merge, but would like changes to marked lines. If those three places are just for testing - but still required, then can we add a #define so I can merge to master and update piMan-zero code to match extra sensors.

micromouseonline committed:
Remove redundant function
update_sensors_control no longer needed
because sensors are updtaed in systick event

micromouseonline committed:
Eliminate loop and delay in sensor report

micromouseonline committed:
Update ecomments in sensor display

micromouseonline committed:
Update comments in setupSystick()

micromouseonline committed:
Comment out diagnostic LED flashing in systick.

Leave the statements and the associated description there
as a reminder that the LED (or some other pin) can be
used to check processor load and timing here.

micromouseonline commented:
Changes made.

Also, a little tidying up.

robzed commented:
Merged
